lib/agent.js

```diff
@@ -190,21 +193,23 @@ Agent.prototype.captureError = function (err, opts, cb) {
       error.context.response = parsers.getContextFromResponse(res, true)
     }
 
-    if (captureFrameError) {
-      // prepare to add a top frame to the stack trace specifying the location
-      // where captureError was called from. This can make it easier to debug
-      // async stack traces.
-      stackman.callsites(captureFrameError, function (_err, callsites) {
+    if (captureLocation) {
+      // prepare to add a stack trace pointing to where captureError was called
+      // from. This can make it easier to debug async stack traces.
+      stackman.callsites(captureLocation, function (_err, callsites) {
         if (_err) {
-          debug('error while getting capture frame callsites: %s', _err.message)
+          debug('error while getting capture location callsites: %s', _err.message)
         }
 
         var next = afterAll(function (_, frames) {
           // As of now, parseCallsite suppresses errors internally, but even if
           // they were passed on, we would want to suppress them here anyway
 
           if (frames) {
-            frames.shift() // ignore the first frame as it will be this module
+            // In case there isn't any log object, we'll make a dummy message
+            // as the APM Server requires a message to be present if a
+            // stacktrace also present
+            if (!error.log) error.log = {message: 'n/a'}
```

Comment:
As mentioned earlier, this can have undesired side effects in grouping together unrelated errors, if only the `error.log.message` is available for grouping.

Reply:
I'm only setting the dummy message if there is no message to begin with. So it's not going to make the grouping worse. The only time that this occurs is if we're about to send an exception to the APM Server, in which case there will be an `exception.message`, but no `log.message`. That's the only time I'm forced to set `log.message` to a dummy value as the API doesn't allow it to be empty.

Reply:
If you have an idea for a better value, or a different approach all together, I'd be happy to hear it though 😃

Reply:
I don't know if it might make sense to use some information from the first stack trace frame, e.g. the filename or the function?

Generally I am fine with your solution, just wanted to point out what happens server side with it.

Reply:
That's a really interesting approach! It could be similar to how the stack frame looks when printed, e.g.:

- `caught at /Users/watson/code/elastic/apm-dev/foo.js:7:9`
- `caught at user (/Users/watson/code/elastic/apm-dev/foo.js:2:9)`

Reply:
@elastic/apm-ui would you be ok with this?

Reply:
After discussing this with the @elastic/apm-ui team, we came to the conclusion that showing either `n/a` or `caught at ...` would not look great on the errors list page as `log.message` would be prioritized and shown instead of `exception.message` if present.

This leaves us with only one option: Copy the `exception.message` into `log.message`. This however unfortunately means that the error specific page will show the message twice - one time for `log` and another time for `exception`. But it's the best solution we can implement given our constraints.

We need to bring this up again for the next version of the APM Server, to ensure this use-case is supported in the intake API so that we don't have to make up a "fake" `log.message`. cc @elastic/apm-server

Reply:
I'm good with copying the `exception.message` to `log.message` in this case 👍

Reply:
Fixed